### PR TITLE
Chore/1415 TS compiler tightening 

### LINKED ISSUE #1415: [Housekeeping] Tighten typescript compiler checks
**Short description of housekeeping request**
As an easy way to lift our code quality, I would like to add a few options to our tsconfig.json:
`noImplicitAny`
`noUnusedLocals`
`noUnusedParameters`

This will result in a large amount of initial errors, so this will be a large PR. Going forward I believe it is well worth it.

### Kick Off

* [ ] Ensure this issue has been `Tech refined` with @kirbydesign/kirby-guild and is updated with a clear implementation description
  _This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation_
* [ ] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code

* [ ] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
* [ ] Implement/update unit tests
* [ ] Create a draft implementation and push to Github
* [ ] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request

### Code Review

* [ ] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review
  _Remember to add `closes #issueno` to the description of the PR._
* [ ] Once approved, merge feature branch/PR to master

:tada: Celebrate

### PR DESCRIPTION
**Pull request by flemcito**
This PR closes #1415 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement (to existing content)
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

Issue Number: #[replace with issue number]

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/cookbook/src/app/examples/slides-example/example-card-content.ts [hunk 1]
```diff
@@ -39,9 +39,9 @@ export class CookbookExampleCardContentComponent implements OnInit, OnChanges {
...
-  ngOnChanges(changes: SimpleChanges) {
-    // not in use in this example - but gives you access to the index of any slide from the component loaded inside the slide
-    let activeSlide = changes.activeSlideIndex.currentValue || 0;
-    // doSomethingWithTheActiveSlideIndex(activeSlide);
-  }
+  // ngOnChanges(changes: SimpleChanges) {
+  //   not in use in this example - but gives you access to the index of any slide from the component loaded inside the slide
+  //   let activeSlide = changes.activeSlideIndex.currentValue || 0;
+  //   doSomethingWithTheActiveSlideIndex(activeSlide);
+  // }
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/slides-example/example-card-content.ts:46]: Consider removing completely.

File: apps/cookbook/src/app/examples/reorder-list-example/reorder-list-example.component.ts [hunk 2]
```diff
@@ -72,7 +72,7 @@ export class ReorderListExampleComponent {
     ev.complete(ev.parentItem.shadowAccounts);
   }
 
-  trackByFn(index, item) {
+  trackByFn(_index, item) {
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/reorder-list-example/reorder-list-example.component.ts:75]: Not used by any trackBy in component markup. Consider removing completely.

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Very nice change. Only some minor suggestions and a question.

File: libs/designsystem/src/lib/testing/test-helper.ts [hunk 3]
```diff
@@ -2,16 +2,16 @@ import { IonicModule } from '@ionic/angular';
...
-    console.warn = patchedWarn;
-  }
+  // private static muteIonicReInitializeWarning() {
+  //   const originalWarn = console.warn;
+  //   const patchedWarn = (warning: any, ...optionalParams: any[]) => {
+  //     const suppress = `Ionic Angular was already initialized. Make sure IonicModule.forRoot() is just called once.`;
+  //     if (warning !== suppress) originalWarn(warning, ...optionalParams);
+  //   };
+  //   console.warn = patchedWarn;
+  // }
```
[the remark this answers is not in the document]
Comment (flemcito) [reply libs/designsystem/src/lib/testing/test-helper.ts:14]: Ah no - just wanted to make sure it didnt break anything and forgot to delete 👍

File: apps/cookbook/src/app/examples/slides-example/example-card-content.ts (hunk 1 above)
Comment (flemcito) [reply apps/cookbook/src/app/examples/slides-example/example-card-content.ts:46]: Yeah, I was bit unsure what to do about it - I'll nuke it

File: libs/designsystem/src/lib/components/stock-chart/stock-chart.component.ts [hunk 4]
```diff
@@ -44,7 +44,7 @@ export class StockChartComponent {
 
   chart: Highcharts.Chart;
 
-  constructor(private hostElement: ElementRef, @Inject(LOCALE_ID) private locale: string) {
```
[the remark this answers is not in the document]
Comment (flemcito) [reply libs/designsystem/src/lib/components/stock-chart/stock-chart.component.ts:47]: Well, no need really. The ` hostElement` constructor parameter is only accessible within the constructor method body this way. 
The `private` keyword automatically creates a class member, but as we only use `hostElement` in the constructor, it's superfluous.

File: apps/cookbook/src/app/examples/reorder-list-example/reorder-list-example.component.ts (hunk 2 above)
Comment (flemcito) [reply apps/cookbook/src/app/examples/reorder-list-example/reorder-list-example.component.ts:75]: 👍  good catch

File: apps/cookbook/src/app/examples/slides-example/example-card-content.ts (hunk 1 above)
Comment (RasmusKjeldgaard) [reply apps/cookbook/src/app/examples/slides-example/example-card-content.ts:46]: Maybe the right thing to do is to make it a code-snippet in slides in the cookbook? Thoughts?

Comment (flemcito) [reply apps/cookbook/src/app/examples/slides-example/example-card-content.ts:46]: Opted to remove it - I dont think it belongs in the cookbook really. It's more of an Angular thing

File: libs/designsystem/src/lib/testing/test-helper.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (flemcito) [reply libs/designsystem/src/lib/testing/test-helper.ts:14]: reintroduced it, and made  `_init` public to avoid a linter warning

**Review by RasmusKjeldgaard — APPROVED** [review]
Good stuff, approved 🎉

**Pull request merged**